### PR TITLE
Refactored pipeline context so that we can subclass image types for BANZAI NRES.

### PR DESCRIPTION
**Pull request by cmccully**

### CODE REVIEW

File: banzai/context.py [hunk 1]
```diff
@@ -16,11 +16,11 @@ def __eq__(self, other):
 
 
 class PipelineContext(object):
-    def __init__(self, command_line_args, allowed_instrument_criteria, processed_path='/archive/engineering/',
-                 post_to_archive=False, fpack=True, rlevel=91, db_address='mysql://cmccully:password@localhost/test',
-                 log_level='INFO', preview_mode=False, max_tries=5, post_to_elasticsearch=False,
-                 elasticsearch_url='http://elasticsearch.lco.gtn:9200', elasticsearch_doc_type='qc',
-                 elasticsearch_qc_index='banzai_qc', **kwargs):
+    def __init__(self, command_line_args, allowed_instrument_criteria, image_class,
```
Comment (kurczynski) [inline banzai/context.py]: It would be good to parameterize this with environment variables at some point.

Comment (cmccully) [reply banzai/context.py]: All of these parameters can be set from the command line currently. Is there justification to do environment variables as well?

Comment (kurczynski) [reply banzai/context.py]: Nope, that's perfect.

File: banzai/main.py [hunk 2]
```diff
@@ -157,64 +157,64 @@ def process_single_frame(pipeline_context, raw_path, filename, last_stage=None,
...
+        pipeline_context = parse_args(selection_criteria, image_class, extra_console_arguments=extra_console_arguments)
 
         if raw_path is None:
             raw_path = pipeline_context.raw_path
     return pipeline_context, raw_path
 
 
 def make_master_bias(pipeline_context=None, raw_path=None):
-    pipeline_context, raw_path = parse_directory_args(pipeline_context, raw_path, settings.IMAGING_CRITERIA)
+    pipeline_context, raw_path = parse_directory_args(pipeline_context, raw_path, settings.IMAGING_CRITERIA, Image)
```
Comment (turnerm) [inline banzai/main.py]: Maybe move `Image` to `settings.IMAGE_CLASS`?

Comment (cmccully) [reply banzai/main.py]: Done.

**Pull request merged**